caillef/dustzh.lua

```diff
@@ -967,14 +1022,30 @@ weaponsMetatable = {
                 self.currentWeapon = weaponInfo
                 self.maxAmmo = weaponInfo.ammo
                 self.ammo = weaponInfo.ammo
+                self.weaponName = weaponInfo.name
                 self:updateAmmoUI()
+                self:updateNameUI()
                 multi:playerAction("changeWeapon", { id = id })
             end
 			p.weaponId = id
 			if self.templates[weaponInfo.item] then
 				local weapon =  Shape(self.templates[weaponInfo.item])
 				weapon.Pivot = self.templates[weaponInfo.item].Pivot
 				self:_setWeapon(p, weapon, weaponInfo, forceNotFPS)
+
+                -- display the weapon next to the weapon name
+                for _, i in ipairs(displayedWeapon) do
+                    Camera:RemoveChild(i)
+                end
+
+                displayedWeapon = {}
+                ui = require("uikit")
+                local displayedWeapon = ui:createShape(Shape(self.templates[weaponInfo.item]))
+                displayedWeapon.parentDidResize = function()
+                    displayedWeapon.LocalPosition = { 160, Screen.Height / 3, 0 }
+                    displayedWeapon.LocalRotation.Y = 180
```

Comment:
can't we change the size of `displayedWeapon` like this ?

```lua
displayedWeapon.Width = 42
displayedWeapon.Height = 42
```

Reply:
i can change the size like this